synapse/cortex.py

```diff
@@ -783,6 +830,126 @@ class Cortex(s_cell.Cell):
         self._initPushLoop()
         self._initFeedLoops()
 
+    async def _loadExtModel(self):
+
+        self.extprops = await (await self.hive.open(('cortex', 'model', 'props'))).dict()
+        self.extunivs = await (await self.hive.open(('cortex', 'model', 'univs'))).dict()
+        self.exttagprops = await (await self.hive.open(('cortex', 'model', 'tagprops'))).dict()
+
+        for form, prop, tdef, info in self.extprops.values():
+            try:
+                self.model.addFormProp(form, prop, tdef, info)
+            except asyncio.CancelledError as e:
+                raise
+            except Exception as e:
+                logger.warning(f'ext prop ({form}:{prop}) error: {e}')
+
+        for prop, tdef, info in self.extunivs.values():
+            try:
+                self.model.addUnivProp(prop, tdef, info)
+            except asyncio.CancelledError as e:
+                raise
+            except Exception as e:
+                logger.warning(f'ext univ ({name}) error: {e}')
+
+        for prop, tdef, info in self.exttagprops.values():
+            try:
+                self.model.addTagProp(prop, tdef, info)
+            except asyncio.CancelledError as e:
+                raise
+            except Exception as e:
+                logger.warning(f'ext tag prop ({name}) error: {e}')
+
+    async def addUnivProp(self, name, tdef, info):
+
+        # the loading function does the actual validation...
+        if not name.startswith('_'):
+            mesg = 'ext univ name must start with "_"'
+            raise s_exc.BadPropDef(name=name, mesg=mesg)
+
+        if info.get('defval', s_common.novalu) is not s_common.novalu:
+            mesg = 'Ext univ may not (yet) have a default value.'
+            raise s_exc.BadPropDef(name=name, mesg=mesg)
+
+        self.model.addUnivProp(name, tdef, info)
+
+        await self.extunivs.set(name, (name, tdef, info))
+
+    async def addFormProp(self, form, prop, tdef, info):
+
+        if not prop.startswith('_'):
+            mesg = 'ext prop must begin with "_"'
+            raise s_exc.BadPropDef(prop=prop, mesg=mesg)
+
+        if info.get('defval', s_common.novalu) is not s_common.novalu:
+            mesg = 'Ext prop may not (yet) have a default value.'
+            raise s_exc.BadPropDef(prop=prop, mesg=mesg)
+
+        self.model.addFormProp(form, prop, tdef, info)
+        await self.extprops.set(f'{form}:{prop}', (form, prop, tdef, info))
+
+    async def delFormProp(self, form, prop):
+        '''
+        Remove an extended property from the cortex.
+        '''
+        full = f'{form}:{prop}'
+
+        pdef = self.extprops.get(full)
+        if pdef is None:
+            mesg = f'No ext prop named {full}'
+            raise s_exc.NoSuchProp(form=form, prop=prop, mesg=mesg)
+
+        for layr in self.layers.values():
```

Comment:
In a situation where a remote layer has data from a extra model property we want to remove from this cortex, this would prevent the removal of that property from the local cortex.